### DISCUSSION TITLE
Lazyloading `Alpine.data` component JS with IntersectionObserver

### DISCUSSION THREAD

**Discussion by philwolstenholme** [alpinejs/alpine#2917] [4. General]
I'm trying to get my initial JS bundle as small as possible, and only download enough JS to have Alpine register the components that will be initially in view.

I'm thinking of splitting each  `Alpine.data` component into its own module, and using IntersectionObserver to only load the JS for a component once it's about to come into view. 

I've not tried it yet, but I was wondering about using `x-ignore` on the components that will be lazyloaded to avoid Alpine throwing errors when it finds components that it does not have a definition for. When the component was ready to be lazyloaded I could load the JS with the component, then I'd remove the `x-ignore` attribute. I _think_ that DOM mutation would trigger Alpine's MutationObserver which would make the component interactive.

Has anyone done anything like this and have any tips? Or anything I should watch out for if I try this idea?

**Comment by vascoeduardo** [comment]
Hey Phil,
I am trying to lazyload my alpine components as you described above. My problem is that removing the x-ignore attribute seems not to trigger Alpine's MutationObserver. Do you have successfully implemented your approach?

**Comment by philwolstenholme** [comment]
@vascoeduardo no, not yet! You might need to manually tell Alpine to check for new components, I think there's a particular function/method that Alpine provides for that.

**Reply by ekwoka** [reply]
It's `Alpine.initTree(el)`

Make sure it's not an element already evaluated.

**Comment by philwolstenholme** [comment]
I've not tried it yet, but https://alistairshepherd.uk/writing/code-splitting-alpine looks very interesting. I think I'll give this plugin a go rather than roll my own.
